source/common/http/http2/codec_impl.h

```diff
@@ -109,13 +109,21 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
   bool wantsToWrite() override { return nghttp2_session_want_write(session_); }
   // Propagate network connection watermark events to each stream on the connection.
   void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
-    for (auto& stream : active_streams_) {
-      stream->runHighWatermarkCallbacks();
+    if (h2_watermark_improvements_) {
```

Comment:
I guess this is why H/1 upstream will still continue reading after downstream connection buffer is above watermark?

Reply:
yes.